G3-2025-V7-#17457-fix header-dropdown-codeviewer

c22isako opened the pull request:
**What's been changed:**
- Linked the dropdown to its corresponding "button" (parent-child).
- Changed so that the dropdown is displayed when clicked, and hidden when hover is outside menu.
- Positioned dropdown so that it's centered under its "button".
- To avoid menu-items being "warped", made the width a set number.

**How it should work:**
- Dropdown should only be displayed when its corresponding "button" is clicked.
- Dropdown should provide the functionality it was meant for, aka write text inside the markdown.
- Dropdown menu-items should be centered under the "button"
- Dropdown & menu-items should not be affected by different sized resolutions.

For additional information, see connected issue: https://github.com/HGustavs/LenaSYS/issues/17457

---
**How to get to functionality**
select course of your choice (which has code-duggor, ex. Testing Course) -> log in -> choose any code-dugga

-> click on the pencil, which brings up the markdown editor (seen within image1.) 
![image](https://github.com/user-attachments/assets/baed6bca-2643-4bd2-b184-e770b83cd4e2)
(image1)

Within the editor, you will find the icon/button & its dropdown, both seen in image2 which is affected by this PR.
![image](https://github.com/user-attachments/assets/2ae5eeb8-62e8-4acd-a9c9-c1f654ee29e9)
(image2)

b23sebth reviewed (changes requested):
The dropdown looks and functions great, but some elements need to be updated for semantic purposes.

## What's changed:
- Dropdown is now correctly linked to the header.
- Dropdown now displays correctly and is centered.
- Correct heading displays after selection.

## Change needed: 
- Update the `span` elements of headerIcon and headerTypes to `<button>` elements instead.

## Other notes:
Some error messages appear when accessing the codeviewer. This also happens on the group branch so is likely not related to this PR.

![image](https://github.com/user-attachments/assets/7b8d88e6-714b-4c82-86ec-218243fb46ab)

A line appears under headings of size 3. This behaves the same on the `G3-2025-V7` branch so it's probably not related to this PR. If this is not intended behavior, it might be worth investigating in a separate issue.

![image](https://github.com/user-attachments/assets/03901ec5-3a6a-49b4-9711-22f999d4c71e)

c22isako commented:
### Ready for re-review!
**Changes made:**
- Updated the span elements of headerIcon and headerTypes to button-elements.

- Changes within styling had to be made to accommodate the changes, such as:
  - Changed affected span- elements to button.
  - Changed the width of buttons to better fit the window.
  - Changed the headerTypes to have the same width as parent div (removed some hard-coded padding)
    - Dropdown-elements Text is now centered (see image below)
  - Changed width of div to fit the header-text better.
  - Moved header Icon to be seperate from imgIcon, enabling seperate styling to better fit the icon.
  
 - Made some changes to the window layout, since it was affected by the introduction of buttons.
   - Removed padding from the icons, since button-elements introduced it (and affected width of window in mobile view)
   - Made the preview-window smaller, to remove some transformation movement when writing text.

**This is the current look of the dropdown:**

![image](https://github.com/user-attachments/assets/6d2e6b71-fa1a-408a-8c0f-74a791d57e1c)

---
### Additional information:
As mentioned in review above, there seems to be errors present which also appear on the group branch. 
I haven't encountered any errors which may have been caused by my changes. 

Since there has been some changes within the layout, I will include before / after for easier troubleshooting in the future. (Changes marked in red, top is "old", bottom is "new")
![image](https://github.com/user-attachments/assets/1f55a1e3-d7cb-4bbe-a379-92ce19a70b55)

_**Note:**
Whiles testing my changes, I found out that the text doesn't wrap within the preview window. This matches the groupbranch, and so I will make this into a seperate issue to fix._
